api/stacks.ts

```diff
@@ -268,7 +268,7 @@ export async function getNftsData(
   const apiUrl = `${getNetworkURL(network)}/extended/v1/tokens/nft/holdings`;
 
   const response = await axios.get<NftEventsResponse>(apiUrl, {
-    timeout: 30000,
+    timeout: 10000,
```

Comment:
out of curiosity, why the reduction of timeout? most of the APIs call uses API_TIMEOUT_MILLI

Reply:
good point. this could do with more review, but the fetching of the STX collectibles dataset needs some optimisation in general.

things I noticed: 
- 30s timeout feels more like the UI is stuck rather than the data is loading
- react-query has a default to retry 3 times with exponential backoff before displaying error: https://tanstack.com/query/v4/docs/react/guides/query-retries. this could result in a 30s * 3 + 1000ms + 2000ms + 4000ms delay already before the user sees an error
- in addition to the fetch to this endpoint, each STX collection contract is then mapped to a promise.all to fetch collection info, which is another potential 30s * 3 + backoff delay